From e5af1028d0af9184fb4e8fbd2b9fcdf218d690f9 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:29:44 -0400
Subject: [net] openvswitch: avoid cast-qual warning in vport_priv

Message-id: <50971889ad7d60decd2a73728e124cd7bd52b236.1410531783.git.jbenc@redhat.com>
Patchwork-id: 93996
O-Subject: [RHEL7.1 net PATCH 11/50] openvswitch: avoid cast-qual warning in vport_priv
Bugzilla: 1110384
RH-Acked-by: Flavio Leitner <fbl@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110384
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit 07dc0602c5976cfc36ccffe5c6d73234f204d585
Author: Daniele Di Proietto <daniele.di.proietto@gmail.com>
Date:   Mon Feb 3 14:08:29 2014 -0800

    openvswitch: avoid cast-qual warning in vport_priv

    This function must cast a const value to a non const value.
    By adding an uintptr_t cast the warning is suppressed.
    To avoid the cast (proper solution) several function signatures
    must be changed.

    Signed-off-by: Daniele Di Proietto <daniele.di.proietto@gmail.com>
    Signed-off-by: Jesse Gross <jesse@nicira.com>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/openvswitch/vport.h b/net/openvswitch/vport.h
index 47d252e..a4d8db7 100644
--- a/net/openvswitch/vport.h
+++ b/net/openvswitch/vport.h
@@ -172,7 +172,7 @@ void ovs_vport_deferred_free(struct vport *vport);
  */
 static inline void *vport_priv(const struct vport *vport)
 {
- return (u8 *)vport + ALIGN(sizeof(struct vport), VPORT_ALIGN);
+ return (u8 *)(uintptr_t)vport + ALIGN(sizeof(struct vport), VPORT_ALIGN);
 }
 
 /**
-- 
1.7.1